src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/CSharpConvertToRecordRefactoringProvider.cs

```diff
@@ -0,0 +1,623 @@
+
+            var positional = CodeAction.Create(
+                positionalTitle,
+                cancellationToken => ConvertToParameterRecordAsync(document, type, classDeclaration, semanticModel, cancellationToken),
+                positionalTitle);
+            var nested = CodeAction.CodeActionWithNestedActions.Create(string.Format(FeaturesResources.Convert_0_to_record, type.Name), ImmutableArray.Create(positional), isInlinable: false);
+            context.RegisterRefactoring(nested);
+        }
+
+        private static async Task<Document> ConvertToParameterRecordAsync(
+            Document document,
+            INamedTypeSymbol originalType,
+            TypeDeclarationSyntax originalDeclarationNode,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            var codeGenerationService = document.GetRequiredLanguageService<ICodeGenerationService>();
+            var syntaxGenerator = document.GetRequiredLanguageService<SyntaxGenerator>();
+
+            var propertiesToMove = originalDeclarationNode.Members.Where(m => ShouldConvertProperty(m, originalType)).Cast<PropertyDeclarationSyntax>().AsImmutable();
+
+            var modifiedClassTrivia = GetModifiedClassTrivia(propertiesToMove, originalDeclarationNode);
+
+            var propertiesToAddAsParams = propertiesToMove.SelectAsArray(p =>
+                SyntaxFactory.Parameter(
+                    new SyntaxList<AttributeListSyntax>(p.AttributeLists.SelectAsArray(attributeList =>
+                        attributeList.Target == null
+                        // convert attributes attached to the property with no target into "property :" targeted attributes
+                        ? attributeList
+                            .WithTarget(SyntaxFactory.AttributeTargetSpecifier(SyntaxFactory.Token(SyntaxKind.PropertyKeyword)))
+                            .WithoutTrivia()
+                        : attributeList.WithoutTrivia())),
+                    modifiers: default,
+                    p.Type,
+                    p.Identifier,
+                    @default: null));
+
+            // remove converted properties and reformat other methods
+            var membersToKeep = GetModifiedMembersForPositionalRecord(originalType, originalDeclarationNode, semanticModel, propertiesToMove, cancellationToken);
+
+            var changedTypeDeclaration = SyntaxFactory.RecordDeclaration(
+                originalType.TypeKind == TypeKind.Class
+                    ? SyntaxKind.RecordDeclaration
+                    : SyntaxKind.RecordStructDeclaration,
+                originalDeclarationNode.GetAttributeLists(),
+                originalDeclarationNode.GetModifiers(),
+                SyntaxFactory.Token(SyntaxKind.RecordKeyword),
+                originalType.TypeKind == TypeKind.Class
+                    ? default
+                    : SyntaxFactory.Token(SyntaxKind.StructKeyword),
```

Comment:
if you're converting a struct to a record-struct, then use the original struct keyword when moving over.  if you're converting a class to a record-class, copy the trivia from `class` over to `record`.